tests/test_regressions.py

```diff
@@ -23,11 +23,12 @@ def test_file_cache_recognizes_consumed_file_handle(self, url):
 
 def test_getattr_during_gc():
     s = CallbackFileWrapper(None, None)
-    # normal behavior:
-    with pytest.raises(AttributeError):
-        s.x
-
-    # this previously had caused an infinite recursion
-    vars(s).clear()  # gc does this.
-    with pytest.raises(AttributeError):
-        s.x
+    with s._CallbackFileWrapper__buf:
```

Comment:
it seems NamedTemporaryFile only issues a ResourceWarning on windows https://github.com/python/cpython/issues/126639